puffin/src/lib.rs

```diff
@@ -612,6 +715,14 @@ impl Drop for ProfilerScope {
     }
 }
 
+/// A unique id for each scope and `ScopeInfo`.
```

Comment:
You should use proper ref-links. Then the compiler could have told you of your mistake here. It also leads to better docs.

```suggestion
/// A unique id for each scope and [`ScopeDetails`].
```